File: apps/api/src/application/orders/list-orders.ts
import type { OrderRepository } from '@/application/ports/order-repository'
import type { OrderSummary } from '@/domain/order/order'

export class ListOrders {
  constructor(private readonly orders: OrderRepository) {}

  execute(): Promise<OrderSummary[]> {
    return this.orders.findAll()
  }
}

File: apps/api/src/ecommerce_api/application/orders/list_orders.py
from ecommerce_api.application.ports.order_repository import OrderRepository
from ecommerce_api.domain.order import OrderSummary


class ListOrders:
    def __init__(self, orders: OrderRepository) -> None:
        self._orders = orders

    async def execute(self) -> list[OrderSummary]:
        return await self._orders.find_all()
